src/Sentry.Unity/SentryOptionsUtility.cs

```diff
@@ -9,35 +9,105 @@ public static void SetDefaults(SentryUnityOptions options, IApplication? applica
+            options.AttachStacktrace = AttackStackTrace;
+            options.SampleRate = SampleRate;
+
+            options.ReleaseOverride = string.Empty;
+            options.EnvironmentOverride = string.Empty;
+
+            options.EnableOfflineCaching = true;
+
+            options.Debug = Debug;
+            options.DebugOnlyInEditor = DebugOnlyInEditor;
+            options.DiagnosticLevel = DiagnosticLevel;
+        }
+
+        private static bool Enabled = true;
+        private static bool CaptureInEditor = true;
+
+        // 'Optimal' and 'Fastest' don't work on IL2CPP. Forcing 'NoCompression'.
+        private static CompressionLevelWithAuto RequestBodyCompressionLevel = CompressionLevelWithAuto.NoCompression;
+
+        private static bool AttackStackTrace = false;
+        private static StackTraceMode StackTraceMode = default;
+        private static float SampleRate = 1.0f;
+        private static bool IsEnvironmentUser = false;
+
+        private static string Release(IApplication application)
+        {
+            return application.ProductName is string productName
                 && !string.IsNullOrWhiteSpace(productName)
                     ? $"{productName}@{application.Version}"
                     : $"{application.Version}";
+        }
 
-            options.DiagnosticLogger?.LogDebug("option.Release: {0}", options.Release);
+        private static string Environment(IApplication application)
+        {
+            return application.IsEditor ? "editor" : "production";
         }
 
-        private static void SetEnvironment(SentryUnityOptions options, IApplication application)
+        private static bool Debug = true;
+        private static bool DebugOnlyInEditor = true;
+        private static SentryLevel DiagnosticLevel = SentryLevel.Warning;
+
+        private static void TryAttachLogger(SentryUnityOptions options, IApplication application)
         {
-            options.Environment ??= application.IsEditor ? "editor" : "production";
-            options.DiagnosticLogger?.LogDebug("option.Environment: {0}", options.Environment);
+            if (options.DiagnosticLogger is null
+                && options.Debug
+                && (!options.DebugOnlyInEditor || application.IsEditor))
+            {
+                options.DiagnosticLogger = new UnityLogger(options);
+            }
         }
 
-        private static void SetCacheDirectoryPath(SentryUnityOptions options, IApplication application)
+        public static void LogOptions(SentryUnityOptions options)
         {
-            options.CacheDirectoryPath ??= application.PersistentDataPath;
-            options.DiagnosticLogger?.LogDebug("option.CacheDirectoryPath: {0}", options.CacheDirectoryPath);
+            var logger = options.DiagnosticLogger;
+            if (logger == null)
+            {
+                return;
+            }
+
+            // TODO: Add missing logs for options of interest
```

Comment:
Alternatively we could implement `ToString` on the options object and just log out a single message after we process the instance. Instead of logging each value of each property.

```suggestion
            logger.LogDebug("Options: {0}", options);
```